sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventHubClientBuilder.java

```diff
@@ -361,6 +363,18 @@ public EventHubClientBuilder prefetchCount(int prefetchCount) {
         return this;
     }
 
+    /**
+     * Set ObjectSerializer implementation to be used for creating ObjectBatch.
+     *
+     * @param serializer ObjectSerializer implementation
+     *
+     * @return updated builder instance
+     */
+    public EventHubClientBuilder serializer(ObjectSerializer serializer) {
+        this.serializer = serializer;
```

Comment:
Yeah this is a very reasonable question.  Right now the client allows serializer to be null, and will check when you go down a serialization code path that serializer is set.  Otherwise it doesn't care if serializer is set or not.  I don't think this is particularly future proof behavior but the alternative aren't much better:
- setting a pass-through "byte array serializer" that serves no purpose except to keep serializer from being null
- fully typing clients such that serializers must always be set
- not fully typing clients with the following options, e.g. in produce
  - Object-type parameter methods that determine if you're should use a serializer or not, like tryAdd(Object o)
  - method overloads for tryAdd(Object o) or tryAdd(T o) and also tryAdd(EventData e)... which forces you to:
    - allow EventDataBatch to contain both SR objects and hand-crafted EventData... which is a bad idea
    - restrict send modes to _either_  EventData mode or Object mode, based on whether a serializer is set